packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/shippingInformation.js

```diff
@@ -1,50 +1,59 @@
-import React, { useCallback, useState } from 'react';
-
-import Button from '../../Button';
+import React, { Fragment } from 'react';
+import { Edit2 as EditIcon } from 'react-feather';
+import { useShippingInformation } from '@magento/peregrine/lib/talons/CheckoutPage/ShippingInformation/useShippingInformation';
 
 import defaultClasses from './shippingInformation.css';
+import ShippingInformationOperations from './shippingInformation.gql';
+import { mergeClasses } from '../../../classify';
+import Icon from '../../Icon';
+import Card from './card';
+import EditForm from './EditForm';
+import EditModal from './editModal';
 
 const ShippingInformation = props => {
-    const { onSave } = props;
+    const { onSave, propClasses } = props;
+    const talonProps = useShippingInformation({
+        onSave,
+        ...ShippingInformationOperations
+    });
+    const {
+        doneEditing,
+        handleEditShipping,
+        loading,
+        shippingData
+    } = talonProps;
+
+    const classes = mergeClasses(defaultClasses, propClasses);
 
-    // TODO: Replace "doneEditing" with a query for existing data.
-    const [doneEditing, setDoneEditing] = useState(false);
-    const handleClick = useCallback(() => {
-        setDoneEditing(true);
-        onSave();
-    }, [onSave]);
+    const rootClassName = doneEditing
+        ? classes.root
+        : classes['root--editMode'];
 
-    const className = doneEditing
-        ? defaultClasses.container
-        : defaultClasses.container_edit_mode;
+    if (loading) {
+        return <span>Loading...</span>;
+    }
 
-    /**
-     * TODO
-     *
-     * Change this to reflect diff UI in diff mode.
-     */
     const shippingInformation = doneEditing ? (
```

Comment:
It occurs to me that `doneEditing` should really be something like `submittedInitialData` or something else indicative of a difference between the initial edit and subsequent "updates".